topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorMockTest.java

```diff
@@ -109,18 +105,15 @@ public void createMockKube(VertxTestContext context) throws Exception {
         m.put(io.strimzi.operator.topic.Config.FULL_RECONCILIATION_INTERVAL_MS.key, "10000");
         session = new Session(kubeClient, new io.strimzi.operator.topic.Config(m));
 
-        Checkpoint async = context.checkpoint();
         vertx.deployVerticle(session, ar -> {
             if (ar.succeeded()) {
                 deploymentId = ar.result();
                 topicsConfigWatcher = session.topicConfigsWatcher;
                 topicWatcher = session.topicWatcher;
                 topicsWatcher = session.topicsWatcher;
                 metrics = session.metricsRegistry;
-                metrics.forEachMeter(meter -> {
-                    metrics.remove(meter);
-                });
-                async.flag();
+                metrics.forEachMeter(meter -> metrics.remove(meter));
+                context.completeNow();
```

Comment:
Maybe I do not understand how the `context.completeNow` is used. But how can it be that we use it in `@BeforeEach` and in `@AfterEach` as well. That does not seem to make sense to me.

Reply:
Funnily enough, I'm working on a slideshow about these subtleties of the Vertx Junit stuff.

Each method gets its own test context instance - so when `@Test testFoo(VertxTestContext context)` runs, and `@BeforeEach beforeFoo(VertxTestContext context)` and `@AfterEach afterFoo((VertxTestContext context)` run before and after it, they are *not* sharing a test context. 

It's rather subtle and caught me out when I removed usage of checkpoints from an `@AfterEach` but didn't remove the `VertxTestContext` parameter - it caused all the tests to time out because the  context in the after each method wasn't being completed.

Reply:
So, I guess this applies only to the annotated methods where the context is injected? Or also to _regular_ unannotated methods where it is passed as a parameter by other method?